pkg/services/mssql/dbms_registered_provision.go

```diff
@@ -0,0 +1,123 @@
+package mssql
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+)
+
+func (d *dbmsRegisteredManager) GetProvisioner(
+	service.Plan,
+) (service.Provisioner, error) {
+	return service.NewProvisioner(
+		service.NewProvisioningStep("preProvision", d.preProvision),
+		service.NewProvisioningStep("getServer", d.getServer),
+		service.NewProvisioningStep("testConnection", d.testConnection),
+	)
+}
+
+func (d *dbmsRegisteredManager) preProvision(
+	_ context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	pp := instance.ProvisioningParameters
+	return &dbmsInstanceDetails{
+		ARMDeploymentName:          "",
+		ServerName:                 pp.GetString("server"),
+		AdministratorLogin:         pp.GetString("administratorLogin"),
+		AdministratorLoginPassword: service.SecureString(pp.GetString("administratorLoginPassword")), // nolint: lll
+	}, nil
+}
+
+func (d *dbmsRegisteredManager) getServer(
+	ctx context.Context,
+	instance service.Instance,
+) (service.InstanceDetails, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	pp := instance.ProvisioningParameters
+	dt := instance.Details.(*dbmsInstanceDetails)
+	resourceGroup := pp.GetString("resourceGroup")
+	result, err := d.serversClient.Get(
+		ctx,
+		resourceGroup,
+		dt.ServerName,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error getting sql server: %s", err)
+	}
+	expectedVersion :=
+		instance.Service.GetProperties().Extended["version"].(string)
+	if *result.Version != expectedVersion {
```

Comment:
What will happen if the user provides a wrong version server and we don't validate version here?

Reply:
And when will the user probably provides a wrong version server? This service is from registered, will broker create two sql server having different version?

Reply:
V11 SQL Server is deprecated. I touched it for a very short time. It does have some gaps with V12. I know few details. As V12 is also specified in the plan property, we should ensure it here.

Reply:
This is mostly for a long term consideration. When V13 comes, we need to add new plans and deprecate the old plans like what the resource provider does.